packages/pieces/google-sheets/src/lib/actions/update-row.ts

```diff
@@ -28,14 +79,21 @@ export const updateRowAction = createAction({
         if (!sheetName) {
             throw Error("Sheet not found in spreadsheet");
         }
-        if (Array.isArray(values)) {
+
+        const formattedValues = [];
+        for (const key in values) {
+            formattedValues.push(values[key]);
+
+        }
+
+        if (formattedValues.length > 0) {
             const res = await googleSheetsCommon.updateGoogleSheetRow({
                 accessToken: context.propsValue['authentication']['access_token'],
                 rowIndex:  Number(context.propsValue.row_id),
                 sheetName: sheetName,
                 spreadSheetId: context.propsValue['spreadsheet_id'],
                 valueInputOption: ValueInputOption.USER_ENTERED,
-                values: values as string[],
+                values: formattedValues as string[],
```

Comment:
Don't use casting, It's make types useless, instead of change the line above:

`const formattedValues: string[] = [];`